primitives/src/signatures/schnorr.rs

```diff
@@ -174,7 +273,7 @@ where
     P: Parameters<BaseField = F> + Clone,
 {
     /// Key-pair generation algorithm
-    pub fn generate<R: CryptoRng + RngCore>(prng: &mut R) -> KeyPair<P> {
+    pub fn generate<R: Rng>(prng: &mut R) -> KeyPair<P> {
```

Comment:
this triggers me to think if we should replace some `RngCore + CryptoRng` trait bound in our API to just `Rng`?

1. We are not implementing an PRNG ourselves, so we should probably always use `Rng` over `RngCore`, (since the latter is for [backend](https://docs.rs/rand/latest/rand/trait.RngCore.html) implemented by the generator)
2. What are some places where we have to use CSPRNG? probably not in schonrr or transcript?

Reply:
I think we need to carefully go through all the usage in JF. This is not likely to be handled within this PR